src/files/XTCFile.cpp

```diff
@@ -0,0 +1,63 @@
+// Chemfiles, a modern library for chemistry file reading and writing
+// Copyright (C) Guillaume Fraux and contributors -- BSD license
+
+#include "chemfiles/files/XTCFile.hpp"
+
+#include "chemfiles/Error.hpp"
+#include "chemfiles/utils.hpp"
+using namespace chemfiles;
+
+#define STRING_0(x) #x
+#define STRING(x) STRING_0(x)
+#define CHECK(x) check_xtc_error((x), (STRING(x)))
+
+XTCFile::XTCFile(std::string path, File::Mode mode)
+    : File(std::move(path), mode, File::DEFAULT), handle_(nullptr) {
+    const char* openmode;
+    if (mode == File::READ) {
+        openmode = "r";
+        CHECK(read_xtc_header(this->path().c_str(), &natoms_, &nframes_, &offsets_));
+    } else if (mode == File::WRITE || mode == File::APPEND) {
+        throw chemfiles::FileError("XTC format does not support writing yet");
+    }
+
+    handle_ = xdrfile_open(this->path().c_str(), openmode);
+    if (!handle_) {
+        throw file_error("could not open the file at {}", this->path());
+    }
+}
+
+XTCFile::~XTCFile() { xdrfile_close(handle_); }
```

Comment:
This is missing a `free(offsets_)`

Reply:
That's kind of embarrassing. My short term memory is bad =)